apsuite/commisslib/meas_touschek_lifetime.py

```diff
@@ -50,253 +69,163 @@ def __str__(self):
 class MeasTouschekLifetime(_BaseClass):
     """."""
 
-    BPMNAME = 'SI-01M2:DI-BPM'
+    AVG_PRESSURE_PV = 'Calc:VA-CCG-SI-Avg:Pressure-Mon'
+    EXC_TUNE_H = 'SI-Glob:DI-Tune-H'
+    EXC_TUNE_V = 'SI-Glob:DI-Tune-V'
     RFFEAttMB = 0  # [dB]  Multibunch Attenuation
     RFFEAttSB = 30  # [dB] Singlebunch Attenuation
     FILTER_OUTLIER = 0.2  # Relative error data/fitting
 
+    # calibration curves measured during machine shift in 2021/09/21:
+    EXCCURVE_SUMA = [-1.836e-3, 1.9795e-4]
+    EXCCURVE_SUMB = [-2.086e-3, 1.9875e-4]
+    OFFSET_DCCT = 8.4e-3  # [mA]
+
     def __init__(self, isonline=True):
         """."""
-        _BaseClass.__init__(self)
-        self.data = dict(measure=dict(), analysis=dict())
-        self.params = MeasTouschekParams()
+        _BaseClass.__init__(
+            self, params=MeasTouschekParams(), target=self._do_measure,
+            isonline=isonline)
+
+        self._recursion = 0
 
         if isonline:
-            self.devices['bpm'] = BPM(MeasTouschekLifetime.BPMNAME)
+            self.create_bpms()
+            self.devices['trigger'] = Trigger('SI-Fam:TI-BPM')
+            self.devices['event'] = Event('Study')
+            self.devices['evg'] = EVG()
             self.devices['currinfo'] = CurrInfoSI()
             self.devices['egun'] = EGun()
-            self.devices['sofb'] = SOFB(SOFB.DEVICES.SI)
-            self.devices['rfcav'] = RFCav()
-
-    def _create_si_bpms(self):
-        si_bpm_filter = {'sec': 'SI', 'sub': '[0-9][0-9](M|C).*'}
-        bpmsnames = BPMSearch.get_names(si_bpm_filter)
-        bpms = {name: BPM(name) for name in bpmsnames}
-        _ = [bpm.wait_for_connection() for bpm in bpms.values()]
-        return bpms
-
-    def set_si_bpms_attenuation(self, bpms=None, value_att=RFFEAttSB):
+            self.devices['rfcav'] = RFCav(RFCav.DEVICES.SI)
+            self.devices['tune'] = Tune(Tune.DEVICES.SI)
+            self.pvs['avg_pressure'] = PV(MeasTouschekLifetime.AVG_PRESSURE_PV)
+            self.pvs['exc_tunex_sts'] = PV(
+                MeasTouschekLifetime.EXC_TUNE_H+':Enbl-Sts')
+            self.pvs['exc_tunex_sel'] = PV(
+                MeasTouschekLifetime.EXC_TUNE_H+':Enbl-Sel')
+            self.pvs['exc_tuney_sts'] = PV(
+                MeasTouschekLifetime.EXC_TUNE_V+':Enbl-Sts')
+            self.pvs['exc_tuney_sel'] = PV(
+                MeasTouschekLifetime.EXC_TUNE_V+':Enbl-Sel')
+
+    def create_bpms(self):
         """."""
-        if bpms is None:
-            bpms = self._create_si_bpms()
-
-        val_old = dict()
-        for name, bpm in bpms.items():
-            val_old[name] = bpm.rffe_att
+        bpmsnames = BPMSearch.get_names({'sec': 'SI', 'dev': 'BPM'})
+        cbn = self.params.bpmname  # chosen bpm name
```

Comment:
I think we don't need to separate apart the chosen BPM here. All BPMs should be equal up to the point when the measurement starts, in case the user decided to change the chosen BPM after the object creation.